web/components/roadmap/MiniRoadmap.tsx

```diff
@@ -15,32 +16,35 @@ interface Props {
 
 export const MiniRoadmap = (props: Props): ReactElement => {
   const { roadmap } = useRoadmap();
-  const getActiveStepNumber = useCallback(() => {
-    if (!props.activeTaskId) return undefined;
-    return roadmap?.steps.find((step) => {
-      const currTask = step.tasks.map((it) => it.id);
-      if (props.activeTaskId) return currTask.includes(props.activeTaskId);
-    })?.step_number;
-  }, [props.activeTaskId, roadmap?.steps]);
-  const [activeStepNumber, setActiveStepNumber] = useState<number | undefined>(getActiveStepNumber());
-  const [openSteps, setOpenSteps] = useState<number[]>([]);
-  const { userData } = useUserData();
-  const roadmapExists = !!roadmap;
-
-  useEffect(() => {
-    const activeStep = getActiveStepNumber();
-    setActiveStepNumber(activeStep);
-    setOpenSteps(activeStep ? [activeStep] : []);
-  }, [props.activeTaskId, roadmapExists, getActiveStepNumber]);
+  const { userData, update } = useUserData();
 
-  const toggleStep = (stepNumber: number): void => {
-    analytics.event.task_mini_roadmap_step.click.expand_contract();
-    if (openSteps.includes(stepNumber)) {
-      setOpenSteps(openSteps.filter((number) => number !== stepNumber));
-    } else {
-      setOpenSteps([...openSteps, stepNumber]);
-    }
-  };
+  const toggleStep = useCallback(
```

Comment:
since this is a callback, can we name it `on...` as in `onToggleStep` or something? because technically this does not toggle the step open/closed (that happens in the step component), it just updates the userdata when that happens